app/views/mappings/index.html.erb

```diff
@@ -1,3 +1,9 @@
+<% if Mapping.has_directory_permission_errors? %>
+  <div class="alert alert-danger" role="alert">
+    <div class="panel-body">Warning: certain directories owned by <code>bmiadmin</code> have incorrect permissions which will prevent app and dataset sharing.. As <code>bmiadmin</code> run the following to set the correct permissions:</div>
+    <div class="panel-body"><pre><code><%= Mapping.directory_permissions_command %></code></pre></div>
+  </div>
+<% end %>
```

Comment:
We already have code https://github.com/OSC/shiny_admin/blob/db773834b39496aae97b81685503709d1bd0bf2c/app/views/layouts/application.html.erb#L55-L63 to handle displaying warning messages; the Rails way wouuld be to add a warning message to the flash object in the controller. Normally the flash object is so the error message appears after a redirect, but you can use https://guides.rubyonrails.org/action_controller_overview.html#flash-now to have it set when rendering. You won't get the exact look of this code, but it will be close:

```ruby
flash.now[:error] = "Warning: certain directories owned by <code>bmiadmin</code> have incorrect permissions which will prevent app and dataset sharing.. As <code>bmiadmin</code> run the following to set the correct permissions: <pre><code><%= Mapping.directory_permissions_command %></code></pre>".html_safe
```